zug/composable.hpp

```diff
@@ -0,0 +1,191 @@
+//
```

Comment:
Since this is now pretty large, it felt to me worth while moving this from `util.hpp` to a separate file. As a part of this step, I also promoted `composable` to the main namespace (outside of `detail`). This felt a little bit more right, since we have a `make_composable` factory. It felt strange that this would return a `detail::composable`.